src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java

```diff
@@ -154,25 +115,20 @@ private FireboltResultSet(InputStream is, String tableName, String dbName, Integ
 		log.debug("ResultSet created");
 	}
 
-	public static FireboltResultSet empty() {
-		return new FireboltResultSet();
-	}
-
 	public static FireboltResultSet of(QueryResult queryResult) throws SQLException {
 		return new FireboltResultSet(new ByteArrayInputStream(queryResult.toString().getBytes()),
-				queryResult.getTableName(), queryResult.getDatabaseName());
+				queryResult.getTableName(), queryResult.getDatabaseName(), defaultCharBufferSize, false, null, false);
 
 	}
 
-	private BufferedReader createStreamReader(InputStream is, Integer bufferSize, boolean isCompressed) {
+	private BufferedReader createStreamReader(InputStream is, int bufferSize, boolean isCompressed) {
 		InputStreamReader inputStreamReader;
 		if (isCompressed) {
 			inputStreamReader = new InputStreamReader(new LZ4InputStream(is), UTF_8);
 		} else {
 			inputStreamReader = new InputStreamReader(is, UTF_8);
 		}
-		return bufferSize != null && bufferSize != 0 ? new BufferedReader(inputStreamReader, bufferSize)
-				: new BufferedReader(inputStreamReader);
+		return new BufferedReader(inputStreamReader, bufferSize);
 	}
```

Comment:
Reading the BufferedReader documentation, it throws an exception when buffer size is 0. We used to account for this by setting it with no buffer size (using the default) but now we'll get an exception in this case. 
Should we be throwing it?

Reply:
I'll need a bit more explanation. Does setting it in FireboltProperties guarantee we'll never receive 0 here?

Reply:
IMHO yes. Moreover we will never get null there. Take a look on the constructor `BufferedReader(Reader)`. It calls `BufferedReader(Reader, defaultBufferSize)` while `defaultBufferSize=8K`. So,  the only thing I did is the code simplification. We always provide reasonable `bufferSize`.

Reply:
Discussed offline, should never happen. In any case buffer size 0 should fail.